frontend/www/js/omegaup/contest/edit.ts

```diff
@@ -113,23 +114,51 @@ OmegaUp.on('ready', () => {
           problems: this.problems,
           requests: this.requests,
           users: this.users,
-          existingProblems: this.existingProblems,
+          searchResultProblems: this.searchResultProblems,
+          searchResultUsers: this.searchResultUsers,
         },
         on: {
-          'update-existing-problems': (query: string) => {
+          'update-search-result-problems': (query: string) => {
             api.Problem.list({
               query,
             })
               .then((data) => {
-                this.existingProblems = [];
-                data.results.forEach((problem: types.ProblemListItem) => {
-                  this.existingProblems.push({
-                    key: problem.alias,
-                    value: problem.title,
+                // Problems previously added into the contest should not be
+                // shown in the dropdown
+                const addedProblems = new Set(
+                  this.problems.map((problem) => problem.alias),
+                );
+                const searchedProblems = new Set(
+                  data.results.map((problem) => problem.alias),
+                );
+                this.searchResultProblems = [...searchedProblems]
+                  .filter((problem) => !addedProblems.has(problem))
+                  .map((key) => ({ key, value: key }));
```

Comment:
```suggestion
                const searchedProblems = this.searchResultProblems
                   .filter((problem) => !addedProblems.has(problem.alias))
                   .map((problem) => ({ key: problem.alias, value: `${problem.title} (${problem.alias})` }));
```

no es necesario hacer la conversión a conjunto: se puede hacer normalito.

Reply:
Listo. Esto servirá para resolver de paso el bug del issue #5237